src/librustc/ty/structural_impls.rs

```diff
@@ -839,14 +839,17 @@ impl<'tcx> TypeFoldable<'tcx> for ty::ExistentialProjection<'tcx> {
 
 impl<'tcx> TypeFoldable<'tcx> for ty::ProjectionTy<'tcx> {
     fn super_fold_with<'gcx: 'tcx, F: TypeFolder<'gcx, 'tcx>>(&self, folder: &mut F) -> Self {
+        let folded_trait_ref = self.trait_ref(folder.tcx()).fold_with(folder);
```

Comment:
`fold_trait_ref` should be gone now. Oh wait. What you found is an AST folder *facepalms*.
Apologies, you could've done this before the rebase too. Just fold the substs.